templates/dialogs/crew-settings.html

```diff
@@ -0,0 +1,114 @@
+            e.preventDefault();
+            add_role("{{ localize 'SWFFG.Crew.Role.New'}}", skills[0], false, false);
+        });
+
+        // remove a role from the list
+        $(".role_selection").on("click", ".remove_role", function(e) {
+          // done on a form listener to handle dynamically added rows
+          $(this).parent().parent()[0].remove();
+        });
+
+        /***
+         * Add a row to the role listing
+         * @param role_name string - name of the role
+         * @param role_skill string - name of the skill to use with the role
+         * @param use_handling bool - indicates if we should use the vehicle handling in the resulting pool
+         * @param use_weapons bool - indicates if we should present vehicle weapons for selection
+         */
+        function add_role(role_name, role_skill, use_handling, use_weapons) {
+            // convert to HTML
+            if (use_handling) {
+                use_handling = "checked";
+            } else {
+                use_handling = '';
+            }
+            if (use_weapons) {
+                use_weapons = "checked";
+            } else {
+                use_weapons = '';
+            }
+            const row = `
+            <div class="flexrow">
+                <div class="flex1">
+                    <input type="text" name="role_name" value="${role_name}" data-dtype="String"/>
+                </div>
+                <div class="flex1">
+                    <select name="role_skill">
+                        ${build_skill_selector(role_skill)}
+                    </select>
+                </div>
+                <div class="flex0">
+                    <input title="{{ localize 'SWFFG.Crew.Role.Picker.Handling'}}" type="checkbox" name="use_handling" data-dtype="Boolean" ${use_handling}/>
+                </div>
+                <div class="flex0">
+                    <input title="{{ localize 'SWFFG.Crew.Role.Picker.Weapons'}}" type="checkbox" name="use_weapons" data-dtype="Boolean" ${use_weapons}/>
+                </div>
+                <div class="flex0">
+                    <i class="fas fa-trash remove_role"></i>
+                </div>
+            </div>
+            `;
+            $('.role_container').append(row);
+        }
+
+        /***
+         * Builds the dropdown selector for role skill
+         * @param skill string - name of the selected skill
+         * @returns {*[]} array - array of HTML (with the specified skill set as selected)
+         */
+        function build_skill_selector(skill) {
+            let options = [];
+            for (let i = 0; i < skills.length; i++) {
+                if (skill === skills[i]) {
+                    options.push(`<option value="${skills[i]}" selected>${skills[i]}</option>`);
+                } else {
+                    options.push(`<option value="${skills[i]}">${skills[i]}</option>`);
+                }
+            }
+            return options;
+        }
+    });
+</script>
```

Comment:
This is very non-Foundry, all this should be in the application class and called in the appropriate application lifecycle functions, or via handlebars helpers.

Reply:
which would work if it didn't permit the ability to add an arbitrary number of them at any time. I don't think Foundry supports this for settings ("this" being adding additional entries), which is why it's jank. Happy to discuss in DMs if you have more thoughts.